lib/evaluation/markForSubmissionEvaluationHandler.ts

```diff
@@ -0,0 +1,114 @@
+import { Descriptor, InputDescriptor, PresentationDefinition } from '@sphereon/pe-models';
+import { nanoid } from 'nanoid';
+
+import { Status } from '../ConstraintUtils';
+
+import { AbstractEvaluationHandler } from './abstractEvaluationHandler';
+import { EvaluationClient } from './evaluationClient';
+import { HandlerCheckResult } from './handlerCheckResult';
+
+export class MarkForSubmissionEvaluationHandler extends AbstractEvaluationHandler {
+  constructor(client: EvaluationClient) {
+    super(client);
+    this.verifiablePresentation.presentationSubmission = {};
+    this.verifiablePresentation.presentationSubmission.descriptor_map = [];
+    this.verifiablePresentation.presentationSubmission.id = nanoid();
+  }
+
+  public getName(): string {
+    return 'MarkForSubmissionEvaluation';
+  }
+
+  public handle(pd: PresentationDefinition, p: any): void {
+    this.iterateOverInputCandidates(pd, p);
+  }
+
+  //TODO move to utils
+  private iterateOverInputCandidates(pd: PresentationDefinition, inputCandidates: any): void {
+    const props = Object.entries(inputCandidates).filter(
+      (x) => Array.isArray(x[1]) && x[1].length && typeof x[1][0] === 'object'
+    ) as Array<[string, Array<unknown>]>;
+    for (const [key, value] of props) {
+      for (const vc of value.entries()) {
+        this.iterateOverInputDescriptors(pd, vc, key);
+      }
+    }
+  }
+
+  private iterateOverInputDescriptors(pd: PresentationDefinition, vc: [number, unknown], path: string): void {
+    const error = this.results.find(
+      (result) => result.status === Status.ERROR && result.verifiable_credential_path === `$.${path}[${vc[0]}]`
+    );
+    if (error) {
+      this.results.push({
+        ...error,
+        message: 'The input candidate is not eligible for submission',
+      });
+    } else {
+      this.createPresentationSubmission(pd, vc, path);
+    }
+  }
+
+  private createPresentationSubmission(pd: PresentationDefinition, vc: any, path: string) {
+    this.verifiablePresentation.presentationSubmission.definition_id = pd.id;
+    const info = this.results.filter((result) => result.verifiable_credential_path === `$.${path}[${vc[0]}]`);
+    if (!this.verifiablePresentation[`${path}`]) {
+      this.verifiablePresentation[`${path}`] = [];
+    }
+    this.handleInputDescriptors(pd.input_descriptors, vc, info, path);
+  }
+
+  private handleInputDescriptors(
```

Comment:
method can be renamed to something like `addInputDescriptorToResults` ?

Reply:
renamed